### DISCUSSION TITLE
LGTM Security analysis

### DISCUSSION THREAD

**Discussion by Moelf** [JuliaLang/julia#44598] [General]
Per Slack discussion, people may have interests taking a look && fix these:
https://lgtm.com/projects/g/JuliaLang/julia/?mode=list

**Comment by rfourquet** [comment]
Please provide more information on what this linked website does.

**Comment by Moelf** [comment]
I think it's pretty self-explained basically it does security analysis etc. For Julia, it doesn't support Julia code but looks like it picked some "bad/ugly" (working) code in the C code base of Julialang.

**Comment by ViralBShah** [comment]
I'm converting this to a discussion. Someone needs to go through these and see if any of them are meaningful and open specific issues/PRs.

**Reply by Seelengrab** [reply]
A cursory look indicates these may be problematic (possible overflows for allocations, stack allocations in a loop..):

 * https://lgtm.com/projects/g/JuliaLang/julia/snapshot/79dc4784d359cbfdca98c60cc982a98df5bd8621/files/src/dump.c#x603fb1bcc2140a9e:1
 * https://lgtm.com/projects/g/JuliaLang/julia/snapshot/79dc4784d359cbfdca98c60cc982a98df5bd8621/files/src/dump.c#x3624d4656bdaad63:1
 * https://lgtm.com/projects/g/JuliaLang/julia/snapshot/79dc4784d359cbfdca98c60cc982a98df5bd8621/files/src/dump.c#xe9d927bc5c01ece6:1
 * https://lgtm.com/projects/g/JuliaLang/julia/snapshot/79dc4784d359cbfdca98c60cc982a98df5bd8621/files/src/staticdata.c#x88205ac3a8d3dd18:1
 * https://lgtm.com/projects/g/JuliaLang/julia/snapshot/79dc4784d359cbfdca98c60cc982a98df5bd8621/files/src/datatype.c#x4dd7a72f5f6a109f:1

Overall, most of the other alerts seem like a good idea to fix & shouldn't be too involved (except for those `FIXME`s that long for future features). The "lesser" ones are A LOT of empty conditional branches without a comment, which should probably be reviewed & either commented or removed.
